tests/helpers.rs

```diff
@@ -0,0 +1,111 @@
+//! Helpers for tests which exercise the overall application, in particular
+//! the `MDBook` initialization and build/rendering process.
+//!
+//! This will create an entire book in a temporary directory using some
+//! dummy content.
+
+#![allow(dead_code, unused_variables, unused_imports)]
+extern crate tempdir;
+
+use std::path::Path;
+use std::fs::{self, File};
+use std::io::{Read, Write};
+
+use tempdir::TempDir;
+
+
+const SUMMARY_MD: &'static str = "# Summary
```

Comment:
Just an idea for maintainability:

We could make a simple directory with a dummy book like:

```
tests
├── config.rs
├── helpers.rs
├── init.rs
├── jsonconfig.rs
├── rendered_output.rs
├── test-book
│   ├── SUMMARY.md
│   ├── conclusion.md
│   ├── first
│   │   ├── index.md
│   │   └── nested.md
│   └── second.md
├── testing.rs
└── tomlconfig.rs
```

and then include the files like this

```rust
static SUMMARY: &str = include_str!("test-book/SUMMARY.md");
```

It makes it easier to edit and expand the book when we add more tests.